tests/stdlib/tsugar.nim

```diff
@@ -3,186 +3,213 @@ discard """
 x + y = 30
 '''
 """
+import std/[sugar, algorithm, random, sets, tables, strutils]
+
+template main() =
+  block: # `=>`
+    block:
+      let f1 = () => 42
+      doAssert f1() == 42
+
+      let f2 = (x: int) => x + 1
+      doAssert f2(42) == 43
+
+      let f3 = (x, y: int) => x + y
+      doAssert f3(1, 2) == 3
+
+      var x = 0
+      let f4 = () => (x = 12)
+      f4()
+      doAssert x == 12
+
+    block:
+      proc call1(f: () -> int): int = f()
+      doAssert call1(() => 12) == 12
+
+      proc call2(f: int -> int): int = f(42)
+      doAssert call2(x => x) == 42
+      doAssert call2((x) => x) == 42
+      doAssert call2((x: int) => x) == 42
+
+      proc call3(f: (int, int) -> int): int = f(1, 2)
+      doAssert call3((x, y) => x + y) == 3
+      doAssert call3((x, y: int) => x + y) == 3
+      doAssert call3((x: int, y: int) => x + y) == 3
+
+      var a = 0
+      proc call4(f: int -> void) = f(42)
+      call4((x: int) => (a = x))
+      doAssert a == 42
```

Comment:
add this:
```nim
      let f5 = (a:int) => (discard) # simplest proc that returns void
      f5(3)
```
and also add this to runnableExamples in sugar.=> as it's not obvious how to write it

Reply:
So for some reason, `runnableExamples` transforms
```nim
let f = () => (discard)
```
to
```nim
let f = () =>
  discard
```
which doesn't compile... (but that doesn't happen when using for example `echo "test"` instead of `discard`)

Reply:
i see, you're hitting https://github.com/nim-lang/Nim/issues/13491

Reply:
How should i proceed on this one?

Reply:
* add it in runnableExamples but commented out
* add a comment in code: # xxx pending #13491: uncomment runnableExamples
* add a test in tsugar